### PR TITLE
Move delivery body

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: docs/asyncapi.yaml
Comment (Kirsten1822) [inline docs/asyncapi.yaml:1]: There's a few references to 'AHWR' in a few of the examples provided in the asyncapi doc - unsure if these need changed to something that relates more to our service or?

File: test/unit/enrichment/enrich-invoice-line.test.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
 const { SCHEME_CODE } = require('../../mocks/values/scheme-code')
-const { FUND_CODE } = require('../../mocks/values/fund-code')
+const scheme = require('../../mocks/scheme')
```
Comment (Kirsten1822) [inline test/unit/enrichment/enrich-invoice-line.test.js:2]: Move the mocks to below the jest.mock?

File: docs/asyncapi.yaml
Comment (johnwatson484) [reply docs/asyncapi.yaml:1]: It's fine in this case.  The examples used all relate to a payment from AHWR so it's fine.  We could have picked any scheme we support at this point.  Key thing is none of the variables are named AHWR.

File: test/unit/enrichment/enrich-invoice-line.test.js [hunk 2]
```diff
@@ -56,13 +54,25 @@ describe('enrich header', () => {
...
     expect(invoiceLine.convergence).toBe(false)
   })
+
+  test('should retain delivery body if already set', async () => {
+    const originalDeliveryBody = invoiceLine.deliveryBody
+    await enrichInvoiceLine(invoiceLine, scheme)
+    expect(invoiceLine.deliveryBody).toBe(originalDeliveryBody)
+  })
+
+  test('should set delivery body if not already set and delivery body defined', async () => {
```
Comment (Kirsten1822) [inline test/unit/enrichment/enrich-invoice-line.test.js:73]: Maybe change the name of this test to something like 'should set delivery body to defined delivery body is not already set'? Does that make it more clear?

File: test/unit/enrichment/process-invoice-lines.test.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-const { FUND_CODE } = require('../../mocks/values/fund-code')
 const { NET_DESCRIPTION, GROSS_DESCRIPTION } = require('../../mocks/values/description')
 const { SOURCE_SYSTEM } = require('../../mocks/values/source-system')
+const scheme = require('../../mocks/scheme')
```
Comment (Kirsten1822) [inline test/unit/enrichment/process-invoice-lines.test.js:3]: Move these mocks to below the jest.mock?

File: test/unit/enrichment/schemas/header.test.js
Comment (Kirsten1822) [inline test/unit/enrichment/schemas/header.test.js:1]: Could you add a space between the constants and the mocks just to separate them?

File: test/unit/enrichment/enrich-invoice-line.test.js (hunk 2 above)
Comment (johnwatson484) [reply test/unit/enrichment/enrich-invoice-line.test.js:73]: I was replicating the language used for the existing fund code tests for consistency.

File: test/unit/enrichment/schemas/invoice-line.test.js [hunk 4]
```diff
@@ -104,4 +104,14 @@ describe('invoice line schema', () => {
     invoiceLine.convergence = false
     expect(schema.validate(invoiceLine)).toBeTruthy()
   })
+
+  test('should fail validation if delivery body is missing', () => {
+    delete invoiceLine.deliveryBody
+    expect(schema.validate(invoiceLine).error).toBeDefined()
+  })
+
+  test('should fail validation if delivery body is incorrect format', () => {
```
Comment (Kirsten1822) [inline test/unit/enrichment/schemas/invoice-line.test.js:113]: Could we change the name of this test to be consistent with the other test file? - 'should fail validation if delivery body is not a valid format'

File: test/unit/enrichment/enrich-invoice-line.test.js (hunk 2 above)
Comment (Kirsten1822) [reply test/unit/enrichment/enrich-invoice-line.test.js:73]: Yeah that makes sense - that's fine then

File: test/unit/enrichment/process-invoice-lines.test.js [hunk 5]
```diff
@@ -1,13 +1,14 @@
-const { FUND_CODE } = require('../../mocks/values/fund-code')
-const { NET_DESCRIPTION, GROSS_DESCRIPTION } = require('../../mocks/values/description')
-const { SOURCE_SYSTEM } = require('../../mocks/values/source-system')
```
Comment (Kirsten1822) [inline test/unit/enrichment/process-invoice-lines.test.js:1]: Delete this space

File: test/unit/enrichment/process-invoice-lines.test.js [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
 jest.mock('../../../app/enrichment/validate-invoice-line')
 const mockValidateInvoiceLine = require('../../../app/enrichment/validate-invoice-line')
 
+const { NET_DESCRIPTION, GROSS_DESCRIPTION } = require('../../mocks/values/description')
+const { SOURCE_SYSTEM } = require('../../mocks/values/source-system')
+const scheme = require('../../mocks/scheme')
+
 const processInvoiceLines = require('../../../app/enrichment/process-invoice-lines')
 
 const invoiceLine = require('../../mocks/payment-requests/invoice-line')
```
Comment (Kirsten1822) [inline test/unit/enrichment/process-invoice-lines.test.js:14]: If this is another mock maybe move it, and it's associated bits, above processInvoiceLines

File: test/unit/enrichment/process-invoice-lines.test.js (hunk 5 above)
Comment (johnwatson484) [reply test/unit/enrichment/process-invoice-lines.test.js]: Killed

**Pull request merged**